File: app/src/main/java/com/digzdigital/divinitytoday/data/db/DbHelper.java
package com.digzdigital.divinitytoday.data.db;

import com.digzdigital.divinitytoday.data.model.Devotional;

import java.util.ArrayList;


public interface DbHelper {
    void setDbListener(DbListener listener);

    void savePost(Devotional devotional);

    void queryForPosts();

    void deletePost(Devotional devotional);

    interface DbListener {
        void onPostsLoaded(ArrayList<Devotional> devotionals);

        void onError(String error);
    }
}

File: app/src/main/java/com/digzdigital/divinitytoday/data/db/DbHelper.kt
package com.digzdigital.divinitytoday.data.db

import com.digzdigital.divinitytoday.data.model.Devotional
import io.reactivex.Observable

interface DbHelper {

    fun savePost(devotional: Devotional): Boolean

    fun queryForPosts(): Observable<List<Devotional>>

    fun deletepost(devotional: Devotional)

}
